### PR TITLE
Derive more things

### PR DESCRIPTION
**Pull request by LoganDark**
I find myself running into unimplemented traits all the time, here are the main offenders. Especially important are Info and GlyphPosition

Let me know if any of the defaults I've picked are inappropriate.

(Also PR 69; nice)

### CODE REVIEW

File: src/woff2.rs [hunk 1]
```diff
@@ -62,7 +63,7 @@ pub struct Woff2TableProvider {
     tables: HashMap<u32, Box<[u8]>>,
 }
 
-#[derive(Debug)]
+#[derive(Copy, Clone, Eq, PartialEq, Debug)]
```
Comment (wezm) [inline src/woff2.rs:66]: This one feels a bit large to be `Copy` (at least 42 bytes).

**Review by wezm — CHANGES_REQUESTED** [review]
Thanks for this, we certainly erred on the side of only deriving what we needed when building Allsorts so there's certainly scope for improvement.

We're fine with `Clone`, `Debug`, and `PartialEq`/`Eq` but would like to request some changes to the others:

- We would prefer to not have the error types be `Copy` in order to reserve the possibility of adding a non-Copy variant in the future.
- We're not huge fans of `Default` unless there's a very clear default value like an empty collection. So I think we'd prefer to avoid this in most cases. The `impl` for `Features seems reasonable though.

**Comment by LoganDark** [comment]
I went ahead and added `#[non_exhaustive]` to the error types to make adding new variants a non-breaking change. That makes this PR a breaking change, but it's good for API stability long-term. (It prevents users from using `match` without a catch-all `_ =>` case.)

You may also want to look into seeing which structs could also be made `#[non_exhaustive]`, which prevents downstream users from constructing them without using `..existing_struct`, so that adding new (public or non-public) fields is a non-breaking change as well.

**Comment by LoganDark** [comment]
Checks have passed

File: src/glyph_position.rs [hunk 2]
```diff
@@ -30,7 +30,7 @@ where
 }
 
 /// The position and advance of a glyph.
-#[derive(Debug, Default, Copy, Clone)]
+#[derive(Copy, Clone, Eq, Debug, Default)]
```
Comment (wezm) [inline src/glyph_position.rs:33]: Probably drop `Default` from this one too.

**Review by wezm — CHANGES_REQUESTED** [review]
Looking good

File: src/glyph_position.rs (hunk 2 above)
Comment (LoganDark) [reply src/glyph_position.rs]: `Default` was there to begin with, but alright

**Comment by LoganDark** [comment]
Looks like that Default was actually needed by allsorts, I didn't add it in this PR so maybe I shouldn't remove it either

**Comment by wezm** [comment]
Oh well this is embarrassing, looks like that `Default` impl I asked you to remove was there prior to your changes and the code was relying on it. Feel free to drop that last commit and accept my apologies for the run around.

**Comment by LoganDark** [comment]
Checks have already passed so merge when ready

**Review by wezm — APPROVED** [review]

**Comment by wezm** [comment]
I merged this and #71 manually. In the end I decided to drop `non_exhaustive` from the error types. I think adding this attribute makes sense when a project has reached a 1.0 release but until then I think it's better to get an error about an unhanded variant if a new one is added, versus it silently being swallowed by a catch all branch in a `match`.

**Pull request closed without merging**